asllib/Typing.ml

```diff
@@ -1735,8 +1726,22 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
         let pat' = best_effort pat (annotate_pattern e env t_e2) in
         (T_Bool |> here, E_Pattern (e2, pat') |> here) |: TypingRule.EPattern
     (* End *)
+    (* Begin EGetArray *)
+    | E_GetArray (e', e_index) -> (
+        let t_e', e'' = annotate_expr_ ~forbid_atcs env e' in
```

Comment:
it's hard to remember which type corresponds to the array element and which one corresponds to the index.
How about e' -> e_base, 
t_e' -> t_base
struct_t_e' -> t_anon_base
ty' -> t_elem